### PR TITLE
task(fxa-payments-server) Add email validation

### PR DESCRIPTION
**Pull request by meandavejustice**
## Because

- We need to validate the new user email

## This pull request

- Adds validation checks to the email input on new user form
- Adds a check to ensure the confirm input matches the original email value

## Issue that this pull request solves

Closes: #9354

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

Luckily our email validation method was in `fxa-shared` which saved a lot of time.

### CODE REVIEW

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx [hunk 1]
```diff
@@ -26,6 +30,8 @@ export const NewUserEmailForm = ({
     middleware: validatorMiddlewareReducer,
   });
 
+  const [firstFieldState, setFirstFieldState] = useState<string>();
```
Comment (dannycoates) [inline packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx:33]: Let's use names that are more descriptive than `firstField` and `secondField`

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx [hunk 2]
```diff
@@ -96,4 +104,57 @@ export const NewUserEmailForm = ({
...
+  }
+  const errorMsg = getString
+    ? /* istanbul ignore next - not testing l10n here */
+      getString('new-user-email-validate')
+    : 'Email is not valid';
+
+  let error = null;
+  if (!valid && !focused) {
+    if (errorMsg) error = errorMsg;
+  }
```
Comment (dannycoates) [inline packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx:127]: nit: we don't need 2 if's here

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx [hunk 3]
```diff
@@ -96,4 +104,57 @@ export const NewUserEmailForm = ({
...
+  value: string,
+  focused: boolean,
+  firstFieldState: string | undefined,
+  getString: Function
+) {
+  let valid = false;
+
+  if (firstFieldState === value) {
+    valid = true;
+  }
```
Comment (dannycoates) [inline packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx:146]: nit `const valid = firstFieldState === value`

**Review by dannycoates — APPROVED** [review]
LGTM. Just some style nits

**Comment by meandavejustice** [comment]
Thank You @dannycoates

**Pull request merged**